soh/soh/Enhancements/tts/tts.cpp

```diff
@@ -228,6 +228,68 @@ void RegisterOnKaleidoscopeUpdateHook() {
             return;
         }
 
+        // Game over + prompts
+        if (pauseCtx->state >= 0xC && pauseCtx->state <= 0x10) {
+            // Reset prompt tracker after state change
+            if (prevState != pauseCtx->state) {
+                prevPromptChoice = -1;
+            }
+
+            switch (pauseCtx->state) {
+                // Game over in full alpha
+                case 0xC: {
```

Comment:
Some docs here would be beneficial eventually. Looks like decomp has a start in `pause.h`
```
typedef enum {
    /*  0 */ PAUSE_STATE_OFF,
    /*  1 */ PAUSE_STATE_WAIT_LETTERBOX, // Request no letterboxing and wait for it.
    /*  2 */ PAUSE_STATE_WAIT_BG_PRERENDER, // Wait for the pause background prerender to be done.
    /*  3 */ PAUSE_STATE_INIT, // Load data and initialize/setup various things.
    /*  4 */ PAUSE_STATE_OPENING_1, // Animate the pause menu coming together with rotations and other animations.
    /*  5 */ PAUSE_STATE_OPENING_2, // Finish some animations for opening the menu.
    /*  6 */ PAUSE_STATE_MAIN, // Pause menu ready for player inputs.
    /*  7 */ PAUSE_STATE_SAVE_PROMPT,  // Save prompt in the pause menu
    /*  8 */ PAUSE_STATE_8,
    /*  9 */ PAUSE_STATE_9,
    /* 10 */ PAUSE_STATE_10,
    /* 11 */ PAUSE_STATE_11,
    /* 12 */ PAUSE_STATE_12,
    /* 13 */ PAUSE_STATE_13,
    /* 14 */ PAUSE_STATE_14,
    /* 15 */ PAUSE_STATE_15,
    /* 16 */ PAUSE_STATE_16,
    /* 17 */ PAUSE_STATE_17,
    /* 18 */ PAUSE_STATE_CLOSING, // Animate the pause menu closing
    /* 19 */ PAUSE_STATE_RESUME_GAMEPLAY // Handles returning to normal gameplay once the pause menu is visually closed
} PauseState;
```

Reply:
yup, I'm waiting for the kaleido docs to get finished which will make this file more manageable to read :)